src/main/java/games/strategy/engine/framework/headlessGameServer/AvailableGames.java

```diff
@@ -75,11 +80,30 @@ String getGameFilePath(final String gameName) {
   private static void populateAvailableGames(final Map<String, URI> availableGames,
       final Set<String> availableMapFolderOrZipNames, final Set<String> mapNamePropertyList) {
     System.out.println("Parsing all available games (this could take a while). ");
-    for (final File map : allMapFiles()) {
-      if (map.isDirectory()) {
-        populateFromDirectory(map, availableGames, availableMapFolderOrZipNames, mapNamePropertyList);
-      } else if (map.isFile() && map.getName().toLowerCase().endsWith(ZIP_EXTENSION)) {
-        populateFromZip(map, availableGames, availableMapFolderOrZipNames, mapNamePropertyList);
+    final List<File> files = allMapFiles();
+    if (files.size() > 0) {
+      final ExecutorService service = Executors.newWorkStealingPool(files.size());
```

Comment:
This is creating one thread per installed map.  I guess that could be fairly large for a bot (~135?).  Just to be safe, should we limit the maximum number of threads that can be created during this operation?

Map parsing is probably a predominantly CPU-bound operation, but there will be some portion that is IO-bound due to reading the fairly-large zip files from disk.  Without going through the trouble of calculating a [blocking coefficient](http://baddotrobot.com/blog/2013/06/01/optimum-number-of-threads/) for this task, I'm wondering if `# of cores + fixed #` would be sufficient.  `fixed #` would be a guessing game just to keep the CPU busy while the next map is being read.  Without measuring, I'm thinking `fixed #` is probably between 1-2.

Reply:
What's your suggestion?

Reply:
You could try both `Runtime.getRuntime().availableProcessors() + 1` and `Runtime.getRuntime().availableProcessors() + 2` with a large number of maps and see which one yields better average performance.